client/src/pages/Feed.tsx

```diff
@@ -0,0 +1,132 @@
+import { Box, Image } from '@chakra-ui/react';
+import { useState, useEffect } from 'react';
+import { z } from 'zod';
+import { MockPosts, PostObjectType, PostObject } from '../components/community/post/MockPosts';
+import { Post } from '../components/community/post/index';
+import customFetch from '../utils/customFetch';
+import { useQuery } from '@tanstack/react-query';
+import { Player } from '@lottiefiles/react-lottie-player';
+
```

Comment:
You could add another function called loader. Loader is a component of react router dom that is the same use as useEffect with [] as empty dependency array. Usually when using with react query, we use it for ensure the query data meaning that we check if the data for this page is cached or not, if it is cached then no fetch and vice versa. You can check my Meal page.

Reply:
> You could add another function called loader. Loader is a component of react router dom that is the same use as useEffect with [] as empty dependency array. Usually when using with react query, we use it for ensure the query data meaning that we check if the data for this page is cached or not, if it is cached then no fetch and vice versa. You can check my Meal page.

You can check out the Loader function at [the latest commit](https://github.com/hungngodev/Flavorie/pull/89/commits/c7bd0b0a73d5435f63fe74c3b31740ddb2b18408), I will fix the conflict with the app file once everything is okay